pylint/extensions/private_import.py

```diff
@@ -0,0 +1,118 @@
+"""Check for use of for loops that only check for a condition."""
+import os
+from typing import TYPE_CHECKING, List
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages, is_node_in_typing_guarded_import_block
+from pylint.interfaces import IAstroidChecker
+
+if TYPE_CHECKING:
+    from pylint.lint.pylinter import PyLinter
+
+
+class PrivateImportChecker(BaseChecker):
+
+    __implements__ = (IAstroidChecker,)
+    name = "import-private-name"
+    msgs = {
+        "C0416": (
+            "Imported private %s (%s)",
+            "import-private-name",
+            "Used when a private module or object prefixed with _ is imported",
+        ),
+    }
+
+    @check_messages("import-private-name")
+    def visit_import(self, node: nodes.Import) -> None:
+        self._check_import_private_name(
+            node, [name[0] for name in node.names], checking_objects=False
+        )
+
+    @check_messages("import-private-name")
+    def visit_importfrom(self, node: nodes.ImportFrom) -> None:
```

Comment:
```suggestion
    def visit_importfrom(self, node: nodes.ImportFrom) -> None:
        if is_node_in_typing_guarded_import_block(node):
            return
```
That way we do the check only once if ``not private_module`` is true.